File: famous-algorithms/kadanes-algorithm/java/KadaneAlgo.java
package com.basicsstrong.algorithm.famousalgorithm;

public class KadaneAlgo{

	public static void main(String[] args) {
		int[] arr = { -2, -1, 3, -2, 4, 3, -3, 5 };
		
		int maxSum = maxSumSubarray(arr);
		
		System.out.println(maxSum);

	}

	private static int maxSumSubarray(int[] arr) {
		
		int globalMax = arr[0];
		int localMax = arr[0];
		
		for(int i = 1; i < arr.length ; i++) {
			
			localMax = Math.max(localMax + arr[i], arr[i]);
			
			globalMax = Math.max(globalMax, localMax);
			
		}
		
		return globalMax;
	}

}

File: famous-algorithms/kadanes-algorithm/javascript/KadaneAlgo.js

function maxSumSubarray(list) {
    
    let globalMax = list[0];
    let localMax = list[0];
    
    for(let i = 1; i < list.length ; i++) {
        
        localMax = Math.max(localMax + list[i], list[i]);
        
        globalMax = Math.max(globalMax, localMax);
        
    }
    
    return globalMax;
}


var list = [-2, -1, 3, -2, 4, 3, -3, 5 ];

const maxSum = maxSumSubarray(list);

console.log(maxSum);
